sys/net/gnrc/network_layer/ipv6/nib/nib_ft.c

```diff
@@ -36,12 +36,25 @@ int gnrc_ipv6_nib_ft_get(const ipv6_addr_t *dst, gnrc_pktsnip_t *pkt,
 
 int gnrc_ipv6_nib_ft_add(const ipv6_addr_t *dst, unsigned dst_len,
                          const ipv6_addr_t *next_hop, unsigned iface,
-                         uint16_t ltime)
+                         uint32_t ltime)
 {
     int res = 0;
     bool is_default_route = ((dst == NULL) || (dst_len == 0) ||
                              ipv6_addr_is_unspecified(dst));
 
+    uint32_t ltime_ms = 0;
+    if (ltime > 0) {
+        /* 0 means infinite lifetime.
+         * The valid lifetime is given in seconds, but our timers work in
+         * milliseconds, so we have to scale down to the smallest possible
+         * value (UINT32_MAX ms). This is however alright since we ask for
+         * a new router advertisement before this timeout expires */
+        ltime_ms = ltime * MS_PER_SEC;
+        if (ltime_ms / MS_PER_SEC != ltime) {
+            /* If there was an overflow take the maximum */
+            ltime_ms = UINT32_MAX;
+        }
+    }
```

Comment:
```suggestion
    uint32_t ltime_ms;
    /* The valid lifetime is given in seconds, but our timers work in
     * milliseconds, so we have to scale down to the smallest possible
     * value (UINT32_MAX ms). This is however alright since we ask for
     * a new router advertisement before this timeout expires */    
    if (ltime > UINT32_MAX / MS_PER_SEC) {
        ltime_ms = UINT32_MAX;
    } else {
        ltime_ms = ltime * MS_PER_SEC;
    }
```